addons/point_of_sale/static/src/js/Popups/ClosePosPopup.js

```diff
@@ -0,0 +1,172 @@
+odoo.define('point_of_sale.ClosePosPopup', function(require) {
+    'use strict';
+
+    const { useState, useRef } = owl.hooks;
+    const AbstractAwaitablePopup = require('point_of_sale.AbstractAwaitablePopup');
+    const Registries = require('point_of_sale.Registries');
+    const { identifyError } = require('point_of_sale.utils');
+    const { ConnectionLostError, ConnectionAbortedError} = require('@web/core/network/rpc_service')
+    const { round_precision: round_pr } = require('web.utils');
+
+    /**
+     * This popup needs to be self-dependent because it needs to be called from different place.
+     */
+    class ClosePosPopup extends AbstractAwaitablePopup {
+        constructor() {
+            super(...arguments);
+            this.manualInputCashCount = true;
+            this.cashControl = this.env.pos.config.cash_control;
+            this.moneyDetailsRef = useRef('moneyDetails');
+            this.closeSessionClicked = false;
+            this.moneyDetails = null;
+            this.state = useState({});
+        }
+        async willStart() {
+            try {
+                const closingData = await this.rpc({
+                    model: 'pos.session',
+                    method: 'get_closing_control_data',
+                    args: [[this.env.pos.pos_session.id]]
+                });
+                this.ordersDetails = closingData.orders_details;
+                this.paymentsAmount = closingData.payments_amount;
+                this.payLaterAmount = closingData.pay_later_amount;
+                this.openingNotes = closingData.opening_notes;
+                this.defaultCashDetails = closingData.default_cash_details;
+                this.otherPaymentMethods = closingData.other_payment_methods;
+
+                // component state and refs definition
+                const state = {notes: '', acceptClosing: false};
+                if (this.cashControl) {
+                    state[this.defaultCashDetails.name] = {counted: 0, difference: -this.defaultCashDetails.amount};
+                }
+                if (this.otherPaymentMethods.length > 0) {
+                    this.otherPaymentMethods.forEach(pm => {
+                        state[pm.name] = {counted: pm.amount, difference: 0}
+                    })
+                }
+                Object.assign(this.state, state);
+            } catch (error) {
+                this.error = error;
+            }
+        }
+        /*
+         * Since this popup need to be self dependent, in case of an error, the popup need to be closed on its own.
+         */
+        mounted() {
+            if (this.error) {
+                this.cancel();
+                if (this.error.message instanceof ConnectionLostError) {
```

Comment:
use the identifyError from `point_of_sale.utils` like so.
```suggestion
                if (identifyError(this.error) instanceof ConnectionLostError) {
```

Reply:
Thanks